homeassistant/components/water_heater/__init__.py

```diff
@@ -39,16 +39,24 @@
 STATE_HIGH_DEMAND = 'high_demand'
 STATE_HEAT_PUMP = 'heat_pump'
 STATE_GAS = 'gas'
+STATE_AUTO = 'auto'
+STATE_MANUAL = 'manual'
+STATE_HEATING = 'heating'
```

Comment:
This is going to make the same mistake as climate has.

There is a difference between operation mode and operating mode.